src/main/webapp/script.js

```diff
@@ -19,15 +19,90 @@ function getRecipeInfo() {
   params.append('image', image);
   const request = new Request('/dishAnalysis', {method: "POST", body: params});
   fetch(request).then(response => response.json()).then((recipeListInfoJson) => {
-    var recipeList = JSON.parse(JSON.parse(recipeListInfoJson));
-    const displayRecipeElement = document.getElementById('display-recipes');
-    displayRecipeElement.innerHTML = "";
-    for (recipe of recipeList) {
-      displayRecipeElement.appendChild(createRecipeElement(recipe));
-    }
+    sessionStorage.recipeList = JSON.parse(recipeListInfoJson);
+    window.location.href = "/display.html";
+  });
+}
+
+/** Fetches and then populates nutrition information section of display page with average fat, calories, etc. */
+function createNutritionElements() {
+  fetch('/dishNutrition?dishName='+dishName).then(response => response.json()).then((dish) => {
+    title.setAttribute("data-rotate", dishName);
+
+    var nutritionElement = document.querySelector(".nutrition-info");
+    Object.keys(dish).forEach(function(key) {
+      var node = document.createElement('div');
+      node.className = 'nutrition-element';
+      node.innerText = 'Average' + key + ': ' + dish[key]['value'] + ' ' + dish[key]['units'];
+      nutritionElement.appendChild(node);
+    });
   });
 }
 
+/** Creates text typing animation */
+window.onload = function() {
+  var text_element = document.getElementById('dish');
+  var toRotate = text_element.getAttribute('data-rotate');
+  console.log("---1.----" + toRotate);
+  var period = text_element.getAttribute('data-period');
+  if (toRotate != null) {
+    new TxtRotate(text_element, toRotate, period);
+  }
+}
+
+var TxtRotate = function(el, toRotate, period) {
+  this.toRotate = toRotate;
+  this.el = el;
+  this.loopNum = 0;
+  this.period = parseInt(period, 10) || 2000;
+  this.txt = '';
+  this.tick();
+  this.isDeleting = false;
+};
+
+/* Print, wait the period, then delete */
+TxtRotate.prototype.tick = function() {
+  var i = this.loopNum % this.toRotate.length;
+  var fullTxt = this.toRotate;
+
+  if (this.isDeleting) {
+    this.txt = fullTxt.substring(0, this.txt.length - 1);
+  } else {
+    this.txt = fullTxt.substring(0, this.txt.length + 1);
+  }
+
+  this.el.innerHTML = '<span class="wrap">' + this.txt + '</span>';
+
+  var that = this;
```

Comment:
Do you need to have a variable "that"? It seems that you can just directly use "this" instead.